src/containers/pages/JurisdictionAssignment/AutoSelectJurisdictions/index.tsx

```diff
@@ -0,0 +1,295 @@
+ *  2). pull metadata
+ *  3). pull rootJurisdiction after getting plan in 1
+ *      -> Each of the above calls are needed for this component to work
+ *  4). render drillDown table where, user can select assignments
+ */
+export const AutoSelectView = (props: JurisdictionAssignmentViewFullProps) => {
+  const {
+    plan,
+    serviceClass,
+    fetchPlanCreator,
+    fetchJurisdictionsMetadataCreator,
+    jurisdictionsMetadata,
+    treeFetchedCreator,
+  } = props;
+
+  const { errorMessage, handleBrokenPage, broken } = useHandleBrokenPage();
+  const [step, setStep] = React.useState<string>(TIMELINE_SLIDER_STEP1);
+  const initialLoadingState = !jurisdictionsMetadata || !plan;
+  const { startLoading, stopLoading, loading } = useLoadingReducer(initialLoadingState);
+
+  React.useEffect(() => {
+    const metadataLoadingKey = startLoading('metadata');
+    loadJurisdictionsMetadata(
+      JURISDICTION_METADATA_RISK,
+      OpenSRPService,
+      fetchJurisdictionsMetadataCreator
+    )
+      .then(() => {
+        stopLoading(metadataLoadingKey);
+      })
+      .finally(() => {
+        stopLoading(metadataLoadingKey);
+      })
+      .catch(error => {
+        displayError(error);
+      });
+  }, []);
+
+  usePlanEffect(
+    props.match.params.planId,
+    plan,
+    serviceClass,
+    fetchPlanCreator,
+    handleBrokenPage,
+    stopLoading,
+    startLoading
+  );
+
+  const rootJurisdictionId = useGetRootJurisdictionId(
+    plan,
+    serviceClass,
+    handleBrokenPage,
+    stopLoading,
+    startLoading
+  );
+
+  useGetJurisdictionTree(
+    rootJurisdictionId,
+    startLoading,
+    treeFetchedCreator,
+    stopLoading,
+    handleBrokenPage
+  );
+
+  if (loading()) {
+    return <Ripple />;
+  }
+
+  if (!plan) {
+    return <ErrorPage errorMessage={COULD_NOT_LOAD_PLAN} />;
+  }
+
+  // TODO - can we use the tree here instead.
+  if (!rootJurisdictionId) {
+    return <ErrorPage errorMessage={COULD_NOT_LOAD_JURISDICTION} />;
+  }
+
+  if (broken) {
+    return <ErrorPage errorMessage={errorMessage} />;
+  }
+
+  const sliderProps = {
+    onClickNext: () => setStep(TIMELINE_SLIDER_STEP2),
+    plan,
+    rootJurisdictionId,
+    serviceClass,
+  };
+
+  const jurisdictionTableProps = {
+    autoSelectionFlow: true,
+    currentParentId: props.match.params.parentId,
+    jurisdictionsMetadata,
+    plan,
+    rootJurisdictionId,
+    serviceClass,
+  };
+
+  const breadcrumbProps = {
+    currentPage: {
+      label: plan.title,
+      url: `${AUTO_ASSIGN_JURISDICTIONS_URL}/${plan.identifier}`,
+    },
+    pages: [
+      {
+        label: PLANNING_PAGE_TITLE,
+        url: PLANNING_VIEW_URL,
+      },
+    ],
+  };
+
+  const timelineSliderProps: TimelineSliderProps = {
+    keyOfCurrentStop: step.toString(),
+    stops: [
+      {
+        keys: '1',
+        labelBelowStop: AUTO_TARGET_JURISDICTIONS_BY_RISK,
+        labelInStop: TIMELINE_SLIDER_STOP1_LABEL,
+      },
+      {
+        keys: ['2', '3'],
```

Comment:
Can we please remove the magic strings "1", "2", and "3"?  Using named constants here is probably better.  Unless there is a reason the magic strings should stay put?